src/lib/ffi/ffi_pkey_algs.cpp

```diff
@@ -955,32 +1032,31 @@ int botan_pubkey_x448_get_pubkey(botan_pubkey_t key, uint8_t output[56]) {
 
 int botan_privkey_load_kyber(botan_privkey_t* key, const uint8_t privkey[], size_t key_len) {
 #if defined(BOTAN_HAS_KYBER)
+   if(key == nullptr) {
+      return BOTAN_FFI_ERROR_NULL_POINTER;
+   }
    *key = nullptr;
-   switch(key_len) {
-      case 1632:
-         return ffi_guard_thunk(__func__, [=]() -> int {
-            const Botan::secure_vector<uint8_t> privkey_vec(privkey, privkey + 1632);
-            auto kyber512 = std::make_unique<Botan::Kyber_PrivateKey>(privkey_vec, Botan::KyberMode::Kyber512_R3);
-            *key = new botan_privkey_struct(std::move(kyber512));
-            return BOTAN_FFI_SUCCESS;
-         });
-      case 2400:
-         return ffi_guard_thunk(__func__, [=]() -> int {
-            const Botan::secure_vector<uint8_t> privkey_vec(privkey, privkey + 2400);
-            auto kyber768 = std::make_unique<Botan::Kyber_PrivateKey>(privkey_vec, Botan::KyberMode::Kyber768_R3);
-            *key = new botan_privkey_struct(std::move(kyber768));
-            return BOTAN_FFI_SUCCESS;
-         });
-      case 3168:
-         return ffi_guard_thunk(__func__, [=]() -> int {
-            const Botan::secure_vector<uint8_t> privkey_vec(privkey, privkey + 3168);
-            auto kyber1024 = std::make_unique<Botan::Kyber_PrivateKey>(privkey_vec, Botan::KyberMode::Kyber1024_R3);
-            *key = new botan_privkey_struct(std::move(kyber1024));
-            return BOTAN_FFI_SUCCESS;
-         });
-      default:
-         BOTAN_UNUSED(key, privkey, key_len);
-         return BOTAN_FFI_ERROR_BAD_PARAMETER;
+
+   const auto mode = [](size_t len) -> std::optional<Botan::KyberMode> {
+      if(len == 1632) {
+         return Botan::KyberMode::Kyber512_R3;
+      } else if(len == 2400) {
+         return Botan::KyberMode::Kyber768_R3;
+      } else if(len == 3168) {
+         return Botan::KyberMode::Kyber1024_R3;
+      } else {
+         return {};
+      }
+   }(key_len);
+
+   if(mode.has_value()) {
+      return ffi_guard_thunk(__func__, [=]() -> int {
+         auto kyber = std::make_unique<Botan::Kyber_PrivateKey>(std::span{privkey, key_len}, *mode);
+         *key = new botan_privkey_struct(std::move(kyber));
+         return BOTAN_FFI_SUCCESS;
+      });
+   } else {
+      return BOTAN_FFI_ERROR_BAD_PARAMETER;
```

Comment:
```suggestion
   if(!mode.has_value()) {
      return BOTAN_FFI_ERROR_BAD_PARAMETER;
   }
   
   return ffi_guard_thunk(__func__, [=]() -> int {
      auto kyber = std::make_unique<Botan::Kyber_PrivateKey>(std::span{privkey, key_len}, *mode);
      *key = new botan_privkey_struct(std::move(kyber));
      return BOTAN_FFI_SUCCESS;
   });
```

... there's a parenthesis-fu that will probably break this patch if applied as is.

Reply:
Current seems cleaner to me tbh